test/controllers/CheckYourAnswersControllerSpec.scala

```diff
@@ -214,83 +217,128 @@ class CheckYourAnswersControllerSpec extends SpecBase with MockitoSugar with Sum
-          val result = route(application, request).value
-          val dataRequest = DataRequest(request, testCredentials, vrn, registration, completeUserAnswers)
-          val expectedAuditEvent = ReturnsAuditModel.build(
-            vatReturnRequest, SubmissionResult.Success, Some(vatReturn.reference), Some(vatReturn.paymentReference), dataRequest
-          )
-          val expectedAuditEventForDataEntry = ReturnForDataEntryAuditModel(vatReturnRequest, vatReturn.reference, vatReturn.paymentReference)
-
-          val userAnswersWithEmailConfirmation = completeUserAnswers.copy().set(EmailConfirmationQuery, true).success.value
-          
-          status(result) mustEqual SEE_OTHER
-          redirectLocation(result).value mustEqual CheckYourAnswersPage.navigate(NormalMode, userAnswersWithEmailConfirmation).url
-
-          verify(auditService, times(1)).audit(eqTo(expectedAuditEvent))(any(), any())
-          verify(auditService, times(1)).audit(eqTo(expectedAuditEventForDataEntry))(any(), any())
-          verify(emailService, times(1))
-            .sendConfirmationEmail(eqTo(registration.contactDetails.fullName),
-              eqTo(registration.registeredCompanyName),
-              eqTo(registration.contactDetails.emailAddress),
-              eqTo(totalVatOnSales),
-              eqTo(vatReturnRequest.period)
-            )(any(), any())
-          verify(mockSessionRepository, times(1)).set(eqTo(userAnswersWithEmailConfirmation))
+      "and corrections is toggled on" - {
+
+        "must redirect to the next page" in {
+
+          val answers =
+            emptyUserAnswers
+              .set(SoldGoodsFromNiPage, false).success.value
+              .set(SoldGoodsFromEuPage, false).success.value
+              .set(CorrectPreviousReturnPage, false).success.value
+
+          when(vatReturnConnector.submitWithCorrection(any())(any())) thenReturn Future.successful(Right(vatReturn, correctionPayload))
+          when(emailService.sendConfirmationEmail(any(), any(), any(), any(), any())(any(), any()))
+            .thenReturn(Future.successful(EMAIL_ACCEPTED))
+
+          val app =
+            applicationBuilder(Some(answers))
+              .configure("features.corrections-toggle" -> true)
+              .overrides(
+                bind[VatReturnConnector].toInstance(vatReturnConnector),
+                bind[CorrectionConnector].toInstance(correctionConnector),
+                bind[EmailService].toInstance(emailService)
+              ).build()
+
+          running(app) {
+            val request = FakeRequest(POST, routes.CheckYourAnswersController.onSubmit(period).url)
+            val result = route(app, request).value
+
+            status(result) mustEqual SEE_OTHER
+            redirectLocation(result).value mustEqual routes.ReturnSubmittedController.onPageLoad(period).url
+            verify(vatReturnConnector, times(1)).submitWithCorrection(any())(any())
```

Comment:
Again your call but we could consider adding verifys in for sending email correctly and for vat return submitting. This is a full happy path

Reply:
We have a specific test for that